packages/amplication-build-manager/src/build-runner/build-runner.service.spec.ts

```diff
@@ -117,29 +170,526 @@ describe("BuildRunnerService", () => {
     ).resolves.not.toThrow();
   });
 
-  it("should copy file and/or directories from `jobPath` to `artifactPath`", async () => {
+  describe("copyFromJobToArtifact", () => {
+    it("should copy file and/or directories from `jobPath` to `artifactPath` and return true", async () => {
+      const resourceId = "resourceId";
+      const buildId = "buildId";
+
+      const spyOnCodeGeneratorSplitterServiceExtractBuildId = jest
+        .spyOn(codeGeneratorSplitterService, "extractBuildId")
+        .mockReturnValue(buildId);
+
+      const jobPath = join(
+        configService.get(Env.DSG_JOBS_BASE_FOLDER),
+        buildId,
+        configService.get(Env.DSG_JOBS_CODE_FOLDER)
+      );
+      const artifactPath = join(
+        configService.get(Env.BUILD_ARTIFACTS_BASE_FOLDER),
+        resourceId,
+        buildId
+      );
+
+      const result = await service.copyFromJobToArtifact(resourceId, buildId);
+
+      expect(spyOnCodeGeneratorSplitterServiceExtractBuildId).toBeCalledTimes(
+        1
+      );
+      expect(spyOnFsExtraCopy).toBeCalledWith(jobPath, artifactPath);
+      expect(result).toBe(true);
+      await expect(fsExtra.copy(jobPath, artifactPath)).resolves.not.toThrow();
+    });
+
+    it("should return false when it gets to the catch block", async () => {
+      const resourceId = "resourceId";
+      const buildId = "buildId";
+
+      const spyOnCodeGeneratorSplitterServiceExtractBuildId = jest
+        .spyOn(codeGeneratorSplitterService, "extractBuildId")
+        .mockReturnValue(buildId);
+
+      spyOnFsExtraCopy.mockImplementationOnce(() => {
+        throw new Error("Error");
+      });
+
+      const result = await service.copyFromJobToArtifact(resourceId, buildId);
+
+      expect(spyOnCodeGeneratorSplitterServiceExtractBuildId).toBeCalledTimes(
+        1
+      );
+      expect(spyOnFsExtraCopy).toBeCalledTimes(1);
+      expect(result).toBe(false);
+    });
+  });
+
+  describe("runBuild", () => {
+    it("On code generation request, it should split the build into jobs when the comparison between the current version and the provided version is greater than or equal to 0", async () => {
```

Comment:
Again here, the test title should be easy to understand without opening the code target of the test.
i.e.
```suggestion
    it("runBuild should split the build into jobs when code generator version has support for the new split job functionality", async () => {
```